pkg/s3wrapper/filesystem_xattr_client.go

```diff
@@ -0,0 +1,117 @@
+package s3wrapper
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func NewFilesystemBasedXattrClient(
+	log logrus.FieldLogger,
+	rootDirectory string,
+) *FilesystemBasedXattrClient {
+	return &FilesystemBasedXattrClient{
+		log:           log,
+		rootDirectory: rootDirectory,
+	}
+}
+
+const (
+	filesystemXattrMetaDataDirectoryName = "metadata"
+	delimiter                            = "::"
+)
+
+type FilesystemBasedXattrClient struct {
+	log           logrus.FieldLogger
+	rootDirectory string
+}
+
+func (xattrClient *FilesystemBasedXattrClient) getPathWithKey(path string, key string) string {
+	key = getKeyWithUserAttributePrefix(key)
+	return fmt.Sprintf("%s%s%s", path, delimiter, key)
+}
+
+func (xattrClient *FilesystemBasedXattrClient) IsSupported() (bool, error) {
+	return true, nil
+}
+
+// getMetaFilenameForManifestFile Finds the base of the filename and the directory for the manifest metadata files
+func (xattrClient *FilesystemBasedXattrClient) getMetaFilenamePrefixForManifestFile(path string) (fileName string, directory string) {
+	relativePath := path[len(xattrClient.rootDirectory):]
+	fileName = filepath.Join(xattrClient.rootDirectory, filesystemXattrMetaDataDirectoryName, relativePath)
+	directory = filepath.Dir(fileName)
+	return
+}
+
+func (c *FilesystemBasedXattrClient) SetRootDirectory(path string) {
+	c.rootDirectory = path
```

Comment:
Why do we need to set and get this? It is passed in the `New...` function, and should never be changed.

Reply:
The short answer is that we cannot construct until we know the baseDir and we only know this at runtime for our tests

See here:
https://github.com/openshift/assisted-service/blob/745923e17a5dfab66cb92c8c9fc975aac9008649/pkg/s3wrapper/filesystem_test.go#L44-L58